api/src/db/extensions.test.ts

```diff
@@ -0,0 +1,108 @@
+import { defaultUserEmail, setupServer } from '../../jest.utils';
+import { createUserInput } from '../utils/create-user';
+
+describe('prisma client extensions', () => {
+  setupServer();
+
+  beforeEach(async () => {
+    await fastifyTestInstance.prisma.user.deleteMany({
+      where: { email: defaultUserEmail }
+    });
+  });
+
+  afterAll(async () => {
+    await fastifyTestInstance.prisma.user.deleteMany({
+      where: { email: defaultUserEmail }
+    });
+  });
+
+  describe('updateCount', () => {
+    it('should default to 0', async () => {
+      const user = await fastifyTestInstance.prisma.user.create({
+        data: createUserInput(defaultUserEmail)
+      });
+
+      expect(user).toMatchObject({
+        updateCount: 0
+      });
+    });
+
+    it('should increment by one for updates and creates', async () => {
+      const user = await fastifyTestInstance.prisma.user.create({
+        data: createUserInput(defaultUserEmail)
+      });
+
+      const updateUser = await fastifyTestInstance.prisma.user.update({
+        where: { id: user.id },
+        data: { username: 'any-change' }
+      });
+
+      expect(updateUser).toMatchObject({
+        username: 'any-change',
+        updateCount: 1
+      });
+
+      const _updateManyResult =
+        await fastifyTestInstance.prisma.user.updateMany({
+          where: { id: user.id },
+          // Even no change to values updates the updateCount
+          data: { username: 'any-change' }
+        });
```

Comment:
```suggestion
      await fastifyTestInstance.prisma.user.updateMany({
        where: { id: user.id },
        // Even no change to values updates the updateCount
        data: { username: 'any-change' }
      });
```
Nit: we're only interested in the side-effects.